tests/Utilities/testframework.f90

```diff
@@ -0,0 +1,158 @@
+!> \file
+!> \author Richard Christie
+!> \brief Framework for test logging.
+!>
+!> \section LICENSE
+!>
+!> Version: MPL 1.1/GPL 2.0/LGPL 2.1
+!>
+!> The contents of this file are subject to the Mozilla Public License
+!> Version 1.1 (the "License"); you may not use this file except in
+!> compliance with the License. You may obtain a copy of the License at
+!> http://www.mozilla.org/MPL/
+!>
+!> Software distributed under the License is distributed on an "AS IS"
+!> basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+!> License for the specific language governing rights and limitations
+!> under the License.
+!>
+!> The Original Code is OpenCMISS-Iron Simple Test Framework:
+!>
+!> The Initial Developer of the Original Code is University of Auckland,
+!> Auckland, New Zealand. Portions created by the University of Auckland are
+!> Copyright (C) 2016 by the University of Auckland. All Rights Reserved.
+!>
+!> Contributor(s):
+!>
+!> Alternatively, the contents of this file may be used under the terms of
+!> either the GNU General Public License Version 2 or later (the "GPL"), or
+!> the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+!> in which case the provisions of the GPL or the LGPL are applicable instead
+!> of those above. If you wish to allow use of your version of this file only
+!> under the terms of either the GPL or the LGPL, and not to allow others to
+!> use your version of this file under the terms of the MPL, indicate your
+!> decision by deleting the provisions above and replace them with the notice
+!> and other provisions required by the GPL or the LGPL. If you do not delete
+!> the provisions above, a recipient may use your version of this file under
+!> the terms of any one of the MPL, the GPL or the LGPL.
+!>
+
+MODULE IRON_TEST_FRAMEWORK
+
+  USE KINDS
+  USE OpenCMISS
+  USE, INTRINSIC :: ISO_FORTRAN_ENV, only : output_unit, error_unit
+
+  IMPLICIT NONE
+
+  PRIVATE
+
+  INTERFACE EXPECT_EQ
+    MODULE PROCEDURE EXPECT_EQ_INTG
+  END INTERFACE EXPECT_EQ
+
+  INTERFACE EXPECT_NEAR
+    MODULE PROCEDURE EXPECT_NEAR_DP
+    MODULE PROCEDURE EXPECT_NEAR_SP
+  END INTERFACE EXPECT_NEAR
+
+  CHARACTER(LEN=:), ALLOCATABLE, SAVE :: current_test_name !< The name of the current test.
+  INTEGER(CMISSIntg), SAVE :: test_err = 0
+
+  PUBLIC INITIALISE_TESTS, FINALISE_TESTS, BEGIN_TEST, END_TEST, &
+    & AUTO_USER_NUMBER, EXPECT_EQ, EXPECT_NEAR
+
+CONTAINS
+
+  !> Initialise contents of test framework. Must be done before first use.
+  SUBROUTINE INITIALISE_TESTS()
+    current_test_name = ""
+    test_err = 0
+  END SUBROUTINE INITIALISE_TESTS
```

Comment:
Yes, that was for the routine names and that is fair enough but is there any reason why the rest of the code can't follow the style guide? This file is a little bit funny in that it is technically outside the library and therefor on one hand you could make the argument that because it is on the outside it doesn't need to follow the style guide. On the other hand, I assume that the idea is that this is distributed with the rest of the library code and therefore it should try to be like the rest of the library code and follow the style guide.